Tests/SwiftDocCUtilitiesTests/XCTestCase+LoadingData.swift

```diff
@@ -0,0 +1,44 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import XCTest
+import SwiftDocC
+import SwiftDocCTestUtilities
+
+extension XCTestCase {
+    /// Loads a documentation catalog from an in-memory test file system.
+    ///
+    /// - Parameters:
+    ///   - catalog: The directory structure of the documentation catalog
+    ///   - otherFileSystemDirectories: Any other directories in the test file system.
+    ///   - configuration: Configuration for the created context.
+    /// - Returns: The loaded documentation bundle and context for the given catalog input.
+    func loadBundle(
+        catalog: Folder,
+        otherFileSystemDirectories: [Folder] = [],
+        configuration: DocumentationContext.Configuration = .init()
+    ) throws -> (DocumentationBundle, DocumentationContext) {
+        let fileSystem = try TestFileSystem(folders: [catalog] + otherFileSystemDirectories)
```

Comment:
Repeat the previous comment: Are many of our tests now faster, avoiding actually touching the file system?

Great to see this centralized utility used so often in the tests!

Reply:
Yes, some tests are a few milliseconds faster but the impact on the entire test suite is less than a second.